src/eos-updater-fetch.c

```diff
@@ -864,6 +895,14 @@ check_scheduler (FetchData     *fetch_data,
                  EOS_UPDATER_ERROR, EOS_UPDATER_ERROR_METERED_CONNECTION,
                  "Error fetching update: %s",
                  "Timed out waiting for download to be scheduled");
+  else if (invalidated_error != NULL)
+    {
+      g_autofree gchar *message = g_strdup_printf ("Download scheduler disappeared unexpectedly: %s",
+                                                   invalidated_error->message);
+      g_set_error (error,
+                   EOS_UPDATER_ERROR, EOS_UPDATER_ERROR_METERED_CONNECTION,
+                   "Error fetching update: %s", message);
```

Comment:
Why not build this error in a single call? (No particular need to change it.)

Reply:
In order to keep the `Error fetching update: %s` string separate, and therefore reusable (it’s used in other error messages in the file).